src/components/App.tsx

```diff
@@ -1849,11 +1952,41 @@ export class App extends React.Component<any, AppState> {
     }
   }
 
+  private savePointerDebounced = throttle(
+    (pointerCoords: { x: number; y: number }) => {
+      if (isNaN(pointerCoords.x) || isNaN(pointerCoords.y)) {
+        // sometimes the pointer goes off screen
+        return;
+      }
+      this.socketInitialized &&
+        this.socket &&
+        this.socket.emit("send-mouse-location", this.roomID, pointerCoords);
+    },
+    50,
+  );
+
   private saveDebounced = debounce(() => {
     saveToLocalStorage(elements, this.state);
   }, 300);
 
   componentDidUpdate() {
+    if (this.state.isCollaborating && !this.socket) {
+      this.initializeSocketClient();
+    }
+    const pointerViewportCoords: {
+      [id: string]: { x: number; y: number };
+    } = {};
+    for (const clientId in this.state.remotePointers) {
+      const remotePointerCoord = this.state.remotePointers[clientId];
+      pointerViewportCoords[clientId] = sceneCoordsToViewportCoords(
+        {
+          sceneX: remotePointerCoord.x,
```

Comment:
can this be undefined and throw?

Reply:
I don't think so, if either x or y are invalid we don't save this.